### PR TITLE
Fixed still executing the given class

### PR DESCRIPTION
**Pull request by Jeffrey-H**
Nulled packet after detecting exploit.

### CODE REVIEW

File: log4jfix-bukkit/src/main/java/dev/frankheijden/log4jfix/bukkit/Log4jFixBukkit.java [hunk 1]
```diff
@@ -25,6 +26,8 @@ public void onPacketSending(PacketEvent event) {
 
                 if (PatternChecker.isExploit(message)) {
                     event.setCancelled(true);
+                    PacketContainer packet = new PacketContainer(PacketType.Play.Client.CHAT);
```
Comment (FrankHeijden) [inline log4jfix-bukkit/src/main/java/dev/frankheijden/log4jfix/bukkit/Log4jFixBukkit.java:29]:
```suggestion
                    PacketContainer packet = new PacketContainer(PacketType.Play.Server.CHAT);
```

**Review by FrankHeijden — COMMENTED** [review]
Thanks for your PR, ill commit the wrong packet in onPacketSending and merge this.

**Comment by Jeffrey-H** [comment]
@FrankHeijden the reason I used client is to prevent other plugins braking on an empty chat message.

**Comment by FrankHeijden** [comment]
Wait, you used the Clientbound packet purposefully in a packetlistener for the Serverbound packet? Could you explain why?

**Comment by Jeffrey-H** [comment]
The whole purpose of this type of packet is so that the server does not recognize it as a packet to process within an event.

**Comment by FrankHeijden** [comment]
> The whole purpose of this type of packet is so that the server does not recognize it as a packet to process within an event.

gotcha. did you by any chance test the same functionality client-side? Does switching the packet around there work too?

**Comment by Jeffrey-H** [comment]
I did not test it on the client-side. But I think that would be irrelevant for vanilla clients. On the other hand, this could be different for modded clients.

The best option I think is to flip these around.

**Comment by FrankHeijden** [comment]
Just to clarify, you say cancelling the event in the listener is not enough, the packet must be replaced with an equivalent invalid one so that it does not process this packet "within an event"? Do you have an example where this is relevant or an incompatibility?

**Pull request merged**